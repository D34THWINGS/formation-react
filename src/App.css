.App {
  padding: 6rem 2rem 2rem;
}
