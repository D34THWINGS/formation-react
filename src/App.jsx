import React, { Component } from 'react';
import styled from '@emotion/styled';
import { Global, css } from '@emotion/core';

import Menu from './components/Menu';
import HousesList from './components/HousesList';
import studentsData from './data/students';
import StudentForm from './components/StudentForm';

const StyledApp = styled.div`
  padding: 6rem 2rem 2rem;
`;

const globalStyles = css`
  body {
    margin: 0;
    padding: 0;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
      "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
      sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
`;

class App extends Component {
  state = {
    students: studentsData,
  };

  handleAddStudent = (student) => {
    const { students } = this.state;
    return this.setState({
      students: [...students, {
        ...student,
        id: students.length + 1,
        house: Math.round(Math.random() * 2) + 1,
      }],
    });
  };

  render() {
    const { students } = this.state;
    return (
      <StyledApp className="App">
        <Global styles={globalStyles} />
        <Menu />
        <StudentForm onAddStudent={this.handleAddStudent} />
        <HousesList students={students} />
      </StyledApp>
    );
  }
}

export default App;
