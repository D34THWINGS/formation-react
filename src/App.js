import React, { Component } from 'react';
import './App.css';
import Menu from './components/Menu';
import HousesList from './components/HousesList';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Menu />
        <HousesList />
      </div>
    );
  }
}

export default App;
