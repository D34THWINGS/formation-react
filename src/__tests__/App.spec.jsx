import React from 'react';
import { shallow } from 'enzyme';

import App from '../App';

const students = [{
  id: 1,
  firstName: 'foo',
  lastName: 'bar',
  house: 2,
}];

describe('App', () => {
  it('should render application', () => {
    const wrapper = shallow(<App students={students} />);
    wrapper.setState({ students });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle student add', () => {
    jest.spyOn(Math, 'random').mockImplementation(() => 0.2);
    const wrapper = shallow(<App students={students} />);
    wrapper.setState({ students });
    const newStudent = { firstName: 'foo', lastName: 'bar' };
    wrapper.find('StudentForm').simulate('addStudent', newStudent);
    expect(wrapper.state().students).toEqual([...students, { ...newStudent, id: 2, house: 1 }]);
    Math.random.mockRestore();
  });
});
