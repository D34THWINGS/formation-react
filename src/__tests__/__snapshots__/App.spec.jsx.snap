// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render application 1`] = `
<div
  className="App"
>
  <Menu />
  <StudentForm
    onAddStudent={[Function]}
  />
  <HousesList
    students={
      Array [
        Object {
          "firstName": "foo",
          "house": 2,
          "id": 1,
          "lastName": "bar",
        },
      ]
    }
  />
</div>
`;
